azext_iot/tests/dps/__init__.py

```diff
@@ -108,6 +110,31 @@ def create_dps(self):
                 )
             )
 
+        new_dps = self.cmd(
+            "iot dps show --name {} --resource-group {} ".format(
+                self.entity_dps_name, self.entity_rg
+            )
+        ).get_output_in_json()
+
+        account = self.cmd("account show").get_output_in_json()
+        user = account["user"]
+
+        if user["name"] is None:
+            raise Exception("User not found")
+
+        while True:
```

Comment:
I know this is how it works on the IoT Hub side as well, but we should think about a max attempt threshold ....like after 10 applications of the role assignment and waiting for 10 seconds, we abort and raise an exception....something like that.